src/main/java/org/fundacionjala/virtualassistant/player/spotify/client/SpotifyClient.java

```diff
@@ -0,0 +1,364 @@
+
+                result.append("Artist: ").append(artistName).append("\n");
+                result.append("Album: ").append(albumName).append("\n");
+                result.append("Track: ").append(trackName).append("\n");
+            } else {
+                result.append("Currently playing: ").append(playbackType).append("\n");
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "Error processing player data";
+        }
+
+        return result.toString();
+    }
+
+    @Override
+    public void playCurrentSong() {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        HttpEntity<String> entity = new HttpEntity<>(headers);
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/player/play",
+                HttpMethod.PUT,
+                entity,
+                String.class
+        );
+    }
+
+    public int extractCurrentTrackPosition(String playerData) {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(playerData);
+
+            if (jsonNode.has("progress_ms")) {
+                return jsonNode.get("progress_ms").asInt();
+            }
+
+            return 0;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return 0;
+        }
+    }
+
+    public void pauseSongOnDevice(String trackUri) {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        String requestBody = "{\"uris\": [\"" + trackUri + "\"]}";
+        HttpEntity<String> entity = new HttpEntity<>(requestBody, headers);
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/player/pause",
+                HttpMethod.PUT,
+                entity,
+                String.class
+        );
+    }
+
+    public boolean playNextTrackOnDevice() {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+
+        HttpEntity<String> entity = new HttpEntity<>(headers);
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/player/next",
+                HttpMethod.POST,
+                entity,
+                String.class
+        );
+
+        return response.getStatusCode() == HttpStatus.NO_CONTENT;
+    }
+
+    public boolean playPreviousTrackOnDevice() {
+        RestTemplate restTemplate = new RestTemplate();
```

Comment:
same here, besides the only thing that changes on this calls are the device why you did not abstracted the divide type?